### PR TITLE
Add Iacio Shamir (throw further)

### PR DESCRIPTION
**Pull request by Dennis-0**
Barium shamir for chestplate - Amplify your throwing power

Items thrown with this shamir equipped will have their motion multiplied by 2:
- Ender pearl
- Snowball
- Egg
- Potion (splash and lingering)
- Any item
- Trident

Note: This does not work in creative mode because it uses the `inventory_changed` advancement

### CODE REVIEW

File: gm4_iacio_shamir/data/gm4_iacio_shamir/functions/check_throw.mcfunction [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#run from advancement
```
Comment (BluePsychoRanger) [inline gm4_iacio_shamir/data/gm4_iacio_shamir/functions/check_throw.mcfunction:1]: the header should say which advancement

File: gm4_iacio_shamir/data/gm4_iacio_shamir/functions/check_throw.mcfunction [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+advancement revoke @s only gm4_iacio_shamir:inventory_changed
+
+execute if score @s gm4_ml_iacio_prl matches 1.. as @e[type=minecraft:ender_pearl,distance=..2,sort=nearest,limit=1] at @s run function gm4_iacio_shamir:add_motion
+execute if score @s gm4_ml_iacio_snw matches 1.. as @e[type=minecraft:snowball,distance=..2,sort=nearest,limit=1] at @s run function gm4_iacio_shamir:add_motion
+execute if score @s gm4_ml_iacio_egg matches 1.. as @e[type=minecraft:egg,distance=..2,sort=nearest,limit=1] at @s run function gm4_iacio_shamir:add_motion
+execute if score @s gm4_ml_iacio_lp matches 1.. as @e[type=minecraft:potion,distance=..2,sort=nearest,limit=1] at @s run function gm4_iacio_shamir:add_motion
+execute if score @s gm4_ml_iacio_sp matches 1.. as @e[type=minecraft:potion,distance=..2,sort=nearest,limit=1] at @s run function gm4_iacio_shamir:add_motion
+
+execute as @e[type=item,distance=..2,nbt={Age:0s},limit=1] at @s run function gm4_iacio_shamir:add_motion
```
Comment (BluePsychoRanger) [inline gm4_iacio_shamir/data/gm4_iacio_shamir/functions/check_throw.mcfunction:12]: why don't you check a drop score here?

**Review by BluePsychoRanger — APPROVED** [review]
I like this module and it's well optimized! Why isn't tridents on this list though? I feel like the most useful projectile would be tridents

File: gm4_iacio_shamir/data/gm4_iacio_shamir/functions/init.mcfunction [hunk 3]
```diff
@@ -0,0 +1,12 @@
+scoreboard objectives add gm4_ml_iacio_tri minecraft.used:minecraft.trident
+scoreboard objectives add gm4_ml_iacio_itm minecraft.custom:minecraft.drop
+scoreboard objectives add gm4_ml_iacio_prl minecraft.used:minecraft.ender_pearl
+scoreboard objectives add gm4_ml_iacio_snw minecraft.used:minecraft.snowball
+scoreboard objectives add gm4_ml_iacio_egg minecraft.used:minecraft.egg
+scoreboard objectives add gm4_ml_iacio_sp minecraft.used:minecraft.splash_potion
+scoreboard objectives add gm4_ml_iacio_lp minecraft.used:minecraft.lingering_potion
```
Comment (Bloo-dev) [inline gm4_iacio_shamir/data/gm4_iacio_shamir/functions/init.mcfunction:7]: `ml_` shouldn't be used as a prefix unless code is part of the metallurgy base.  This change needs to be made in multiple sections of code throughout the module.

**Review by Bloo-dev — CHANGES_REQUESTED** [review]
This is a very neat shamir. This is some very optimized and minimal code, well done.
Please resolve the comment I left.

**Review by Bloo-dev — APPROVED** [review]
This will now advance to server testing.

**Review by SandSander — APPROVED** [review]

**Pull request merged**